pose_prediction/plugin.cpp

```diff
@@ -44,7 +63,7 @@ class pose_prediction_impl : public pose_prediction {
 	}
 
 	Eigen::Quaternionf apply_offset(const Eigen::Quaternionf& orientation) const {
-		std::lock_guard<std::mutex> lock {offset_mutex};
+		std::lock_guard<std::recursive_mutex> lock {offset_mutex};
```

Comment:
This mutex is on the critical path of all pose queries. Since pose predict will run on whatever core is calling it, `offset_mutex` will bounce around. I understand this is required only because of a concurrent write to `offset` in `set_offset`, which can only come from debugview. Should we just get rid of that button from debugview and remove the mutex?

All of that being said, there might not be a perf hit at all because pose predict is only called a few hundred times a second, not a few thousand.

Reply:
Resolved by using a `shared_mutex` which has a non-exclusive read state.